### DISCUSSION TITLE
[Python] [FastAPI] Deployment Failed

### DISCUSSION THREAD

**Discussion by VincentLeV** [deta/cloud-docs#663] [Help]
Hi!

I'm trying to deploy my Python Flask app to Micros. The error message is somehow like this

```
Error: failed to update dependencies: error on one or more dependencies, 
no dependencies were added, see output for details.
```
I searched in the Docs and I can see the it's because of the `requirements.txt` not encoded correctly. So I tried to encode it to `UTF-8` but I still couldn't deploy the app. In the terminal, it said `Adding dependencies` already but then there's new error

```
ERROR: Invalid requirement: '\ufeffappdirs==1.4.4'
You should consider upgrading via the 'pip install --upgrade pip' command.


Error: failed to update dependecies: error on one or more dependencies, no dependencies were added, see output for details
```

Could someone help me with this? Thank you!

**Comment by lemonyte** [comment]
You have a zero-width non-breaking space (`\ufeff`) in your requirements.txt file. Try deleting it and creating it again with utf-8 encoding.

**Reply by VincentLeV** [reply]
Thank you for your reply!

How could I find that space in my file? There's an empty line at the end of the file and I deleted it but nothing changed. Then I tried to use notepad to save it as "UTF-8 with BOM" option but it didn't help

**Reply by lemonyte** [reply]
Sorry, I wasn't clear. I meant delete the entire file and create it with utf-8 encoding from scratch. You won't see the character, but it's right before `appdirs==1.4.4`. (Side note, try not to use `pip freeze`, only list the dependencies you actually use)

**Reply by VincentLeV** [reply]
Yes, I did exactly as you said and I don't have anymore the error related to dependencies anymore. But now there's a new error 
```
Adding dependencies...
Error: failed to update dependencies: Internal server error
```
I kept trying many times but nothing changed. I have no idea what's happening here
